libs/wingsdk/src/core/awscdk-app.ts

```diff
@@ -0,0 +1,178 @@
+    const cdkOutdir = join(outdir, "./cdk.out");
+
+    mkdirSync(cdkOutdir, { recursive: true });
+
+    const cdkApp = new cdk.App({ outdir: cdkOutdir });
+    const cdkStack = new cdk.Stack(cdkApp, CDK_STACK_NAME);
+
+    if (!props.customFactory) {
+      throw new Error(
+        "A custom factory must be passed to the base CdkApp class."
+      );
+    }
+    Polycons.register(cdkStack, props.customFactory);
+
+    super(cdkStack, "Default");
+
+    this.outdir = outdir;
+    this.cdkApp = cdkApp;
+    this.cdkStack = cdkStack;
+
+    // register a logger for this app.
+    Logger.register(this);
+  }
+
+  /**
+ * Synthesize the app into CDK configuration in a `cdk.out` directory.
+ *
+ * This method returns a cleaned snapshot of the resulting CDK template
+ * for unit testing.
+ */
+  synth(): string {
+    // synthesize cdk.Stack files in `outdir/cdk.out`
+    this.cdkApp.synth();
+    this.synthPackageJson();
+    this.synthDeployCode();
+    this.synthDestroyCode();
+
+    const template = Template.fromStack(this.cdkStack)
+    return stringify(template.toJSON(), null, 2) ?? "";
+  }
+
+  /**
+   * Create destroy.js file
+   */
+  synthDestroyCode(): void {
+    const code = [
+      "const { intro, outro, spinner } = require(\"@clack/prompts\");",
+      "const { readFileSync, unlinkSync, existsSync } = require(\"fs\");",
+      "const { execSync } = require(\"child_process\");",
+      "const color = require(\"picocolors\");",
+      "",
+      "async function main() {",
+      "  intro(`${color.bgCyan(color.black(\" CDK App \"))}`);",
+      "",
+      "  var appName;",
+      "  if (existsSync(\"./stack-name\")) {",
+      "    appName = readFileSync(\"./stack-name\", { encoding: \"utf8\" });",
+      "  } else {",
+      "    outro(`Can't find the stack name`);",
+      "    return;",
+      "  }",
+      "",
+      "  const s = spinner();",
+      "  s.start(\"Removing stack\");",
+      "  execSync(`aws cloudformation delete-stack --stack-name ${appName}`);",
+      "  s.stop(\"Stack removed\");",
+      "",
+      "  s.start(\"Deleting stack-name file\");",
+      "  unlinkSync(\"./stack-name\");",
+      "  s.stop(\"stack-name file deleted\");",
+      "",
+      "  outro(\"CDK App removed\");",
+      "}",
+      "",
+      "main().catch(console.error);",
+    ];
+
+    writeFileSync(this.outdir.concat("/destroy.js"), code.join("\n"), "utf8");
+  }
+
+  /**
+   * Create deploy.js file
+   */
+  synthDeployCode(): void {
+
+    const code = [
+      "const { intro, outro, spinner, text, note } = require(\"@clack/prompts\");",
+      "const { writeFileSync, readFileSync, existsSync } = require(\"fs\");",
+      "const { execSync } = require(\"child_process\");",
+      "const color = require(\"picocolors\");",
+      "",
+      "async function main() {",
+      "  intro(`${color.bgCyan(color.black(\" CDK App \"))}`);",
+      "",
+      "  var appName;",
+      "  if (existsSync(\"./stack-name\")) {",
```

Comment:
Similarly to terraform, let's just tell users to install the CDK cli and do:

```sh
$ wing compile -t awscdk hello.w
$ cd target/hello.awscdk
$ cdk deploy
```

Reply:
I'm not sure how to use cdk deploy in this case. As far as I know, cdk deploy uses cdk.json to synthesize and deploy a stack. However, when we run wing compile -t awscdk app.w, the cdk.out directory is created with the necessary files to deploy, so we need to find a way to deploy only those files. I'm not sure how to configure cdk.json for that purpose.

The alternative I found was to use cdk-assets to upload the required files to the CDK's S3 folder by using the root.assets.json file, and then execute aws cloudformation deploy with the root.template.json file.

Reply:
`cdk deploy` accepts a `cdk.out` directory in the `--app` option:

```sh
cdk deploy --app path/to/cdk.out
```

Reply:
It works, cool, I will remove the scripts

Reply:
I think I've hit a roadblock.
Using the `cdk deploy` command works, but in the console, all the created stacks have the same name. I tried using `node.addr`, but it doesn't change the stack even if I use a different path.

Reply:
I see... I think you'll need to load the stack name from an environment variable like `CDK_STACK_NAME`

See an example of AZURE_LOCATION.